tools/crm_mon.c

```diff
@@ -852,149 +848,54 @@ set_fencing_options(int level)
     }
 }
 
-/* Before trying to connect to fencer or cib check for state of
-   pacemakerd - just no sense in trying till pacemakerd has
-   taken care of starting all the sub-processes
-
-   Only noteworthy thing to show here is when pacemakerd is
-   waiting for startup-trigger from SBD.
- */
-static void
-pacemakerd_event_cb(pcmk_ipc_api_t *pacemakerd_api,
-                    enum pcmk_ipc_event event_type, crm_exit_t status,
-                    void *event_data, void *user_data)
+static int
+connect_daemons(void)
 {
-    pcmk_pacemakerd_api_reply_t *reply = event_data;
-    enum pcmk_pacemakerd_state *state =
-        (enum pcmk_pacemakerd_state *) user_data;
-
-    /* we are just interested in the latest reply */
-    *state = pcmk_pacemakerd_state_invalid;
+    int rc = pcmk_rc_ok;
+    time_t last_updated = 0;
 
-    switch (event_type) {
-        case pcmk_ipc_event_reply:
-            break;
+    CRM_CHECK(cib != NULL, return EINVAL);
 
-        default:
-            return;
+    if (cib->state != cib_disconnected) {
+        return rc;
     }
 
-    if (status != CRM_EX_OK) {
-        out->err(out, "Bad reply from pacemakerd: %s",
-                 crm_exit_str(status));
-        return;
-    }
+    if (cib->variant == cib_native) {
+        rc = pcmk__pacemakerd_status(out, crm_system_name,
+                                     options.reconnect_ms / 2, false,
+                                     &pcmkd_state);
+        if (rc != pcmk_rc_ok) {
+            return rc;
+        }
 
-    if (reply->reply_type != pcmk_pacemakerd_reply_ping) {
-        out->err(out, "Unknown reply type %d from pacemakerd",
-                 reply->reply_type);
-    } else {
-        if ((reply->data.ping.last_good != (time_t) 0) &&
-            (reply->data.ping.status == pcmk_rc_ok)) {
-            *state = reply->data.ping.state;
+        last_updated = time(NULL);
+
+        switch (pcmkd_state) {
+            case pcmk_pacemakerd_state_running:
+            case pcmk_pacemakerd_state_shutting_down:
+            case pcmk_pacemakerd_state_remote:
+                /* Fencer and CIB may still be available while shutting down or
+                 * running on a Pacemaker Remote node
+                 */
```

Comment:
I do think it's useful to indicate if the cluster is shutting down

Reply:
Sure, easy change.

Reply:
Heh, I was wrong about this :)

Due to the mess of when we can/can't call `out->reset()` and `out->finish()` thanks to the possibility of curses output, yesterday's versoin of this PR was never really showing the `pacemakerd` status at all. It was being called, but the output was being reset or cleared before the buffer was flushed.

I ended up doing something you suggested earlier, that I had decided against at the time: add the `pacemakerd` status to the cluster summary if we have a native CIB connection and get that far (and still output the `pacemakerd` status by itself if we don't get that far). It's the cleanest option I found to implement (the other refactors helped), and the resulting output looks pretty nice IMO. It shows the Pacemaker status even if it's "running," but it doesn't look out of place since it's a summary list item.

```
Cluster Summary:
  * Stack: corosync
  * Pacemaker status: Pacemaker is running
  * Current DC:	node1 (version 2.1.5-0b3656e85) - partition with quorum
  * Last updated: Fri Nov  4 17:35:44 2022
  * Last change:  Fri Nov  4 17:06:38 2022 by root via cibadmin on node1
  * 1 node configured
  * 2 resource instances configured
```

Reply:
What do you think of something like

    * Stack: corosync (Pacemaker is running)

or 

    * Stack: Pacemaker is running with corosync as cluster layer

to conserve screen space? It fits the concept of "stack" which is Pacemaker layered on top of Corosync. We could tweak the pacemaker messages to fit the usage.

Reply:
This would also avoid needing new display options since the stack option would control it. The XML for \<stack\> could include the pacemakerd status as a separate attribute.